src/python/pants/option/config_test.py

```diff
@@ -22,72 +38,175 @@ def setUp(self) -> None:
       list: [1, 2, 3, %(answer)s]
-      listappend: +[7, 8, 9]
+      list2: +[7, 8, 9]
 
       [b]
       preempt: True
+      
+      [b.nested]
       dict: {
           'a': 1,
           'b': %(answer)s,
-          'c': ['%(answer)s', %(answer)s]
+          'c': ['%(answer)s', '%(answer)s'],
         }
+      
+      [b.nested.nested-again]
+      movie: inception
       """
     )
-
-    self.ini2_content = textwrap.dedent(
+    ini2_content = dedent(
       """
       [a]
       fast: True
 
       [b]
       preempt: False
+      
+      [c.child]
+      no_values_in_parent: True
 
       [defined_section]
       """
     )
+    self.config = self._setup_config(ini1_content, ini2_content, suffix=".ini")
+    self.default_seed_values = Config._determine_seed_values(
+      seed_values={"buildroot": self.build_root},
+    )
+    self.default_file1_values = {
+      "name": "foo",
+      "answer": "42",
+      "scale": "1.2",
+      "path": "/a/b/42",
+      "embed": "/a/b/42::foo",
+      "disclaimer": "\nLet it be known\nthat.",
+    }
+    self.expected_file1_options = {
+      "a": {
+        "list": "[1, 2, 3, 42]",
+        "list2": "+[7, 8, 9]",
+      },
+      "b": {
+        "preempt": "True",
+      },
+      "b.nested": {
+        "dict": "{\n'a': 1,\n'b': 42,\n'c': ['42', '42'],\n}"
+      },
+      "b.nested.nested-again": {
+        "movie": "inception",
+      },
+    }
+    self.expected_file2_options: Dict[str, Dict[str, str]] = {
+      "a": {
+        "fast": "True",
+      },
+      "b": {
+        "preempt": "False",
+      },
+      "c.child": {
+        "no_values_in_parent": "True",
+      },
+      "defined_section": {},
+    }
+    self.expected_combined_values: Dict[str, Dict[str, str]] = {
+      **self.expected_file1_options,
+      **self.expected_file2_options,
+      "a": {
+        **self.expected_file2_options["a"], **self.expected_file1_options["a"],
+      },
+    }
 
-    with temporary_file(binary_mode=False) as ini1, \
-      temporary_file(binary_mode=False) as ini2, \
-      temporary_file_path() as buildroot:
-      ini1.write(self.ini1_content)
-      ini1.close()
-      ini2.write(self.ini2_content)
-      ini2.close()
-      self.config = Config.load(
-        config_paths=[ini1.name, ini2.name], seed_values={"buildroot": buildroot}
-      )
-      self.assertEqual([ini1.name, ini2.name], self.config.sources())
-
-  def test_getstring(self) -> None:
-    self.assertEqual('/a/b/42', self.config.get('a', 'path'))
-    self.assertEqual('/a/b/42::foo', self.config.get('a', 'embed'))
-    self.assertEqual('[1, 2, 3, 42]', self.config.get('a', 'list'))
-    self.assertEqual('+[7, 8, 9]', self.config.get('a', 'listappend'))
-    self.assertEqual(
-      """
-Let it be known
-that.""",
-      self.config.get('b', 'disclaimer'))
+  def test_sections(self) -> None:
+    expected_sections = list(
+      OrderedSet([*self.expected_file2_options.keys(), *self.expected_file1_options.keys()])
+    )
```

Comment:
We want to duplicate but still preserve order.

Reply:
Did you mean "de-duplicate"? Just checking!

Reply:
I did! Ooops.